controllers/infinispan_controller.go

```diff
@@ -1760,18 +1765,23 @@ func GossipRouterPodList(infinispan *infinispanv1.Infinispan, kube *kube.Kuberne
 	return podList, kube.ResourcesList(infinispan.Namespace, GossipRouterPodLabels(infinispan.Name), podList, ctx)
 }
 
-func buildStartupArgs(overlayConfigMapKey string, zeroCapacity string) []string {
-	var args []string
+func buildStartupArgs(overlayConfigMapKey, overlayLog4jConfigMapKey, zeroCapacity string) []string {
+	var args = []string{"-Dinfinispan.zero-capacity-node=" + zeroCapacity}
+	if overlayLog4jConfigMapKey != "" {
+		args = append(args, "-l", "user/"+overlayLog4jConfigMapKey)
+	} else {
+		args = append(args, "-l", OperatorConfMountPath+"/log4j.xml")
+	}
 	if overlayConfigMapKey != "" {
-		args = []string{"-Dinfinispan.zero-capacity-node=" + zeroCapacity, "-l", OperatorConfMountPath + "/log4j.xml", "-c", "user/" + overlayConfigMapKey, "-c", "operator/infinispan.xml"}
+		args = append(args, "-c", "user/"+overlayConfigMapKey, "-c", "operator/infinispan.xml")
 	} else {
-		args = []string{"-Dinfinispan.zero-capacity-node=" + zeroCapacity, "-l", OperatorConfMountPath + "/log4j.xml", "-c", "operator/infinispan.xml"}
+		args = append(args, "-c", "operator/infinispan.xml")
```

Comment:
We can make this method more efficient and easier to read by using a [strings.Builder](https://pkg.go.dev/strings#Builder)

Reply:
ok, looking into it.